### ISSUE TITLE
Exporting a type when no TSNamedType is available but a TSType is leads to confusing behavior

### ISSUE THREAD

**Issue by dhoepelman** [scala-tsi/scala-tsi#82]
Split from #74 reported by @tonymichaelson 

An (enumeratum) enum generates different typescript when references directly or as part of another class. This should be unified and de-duplicated.

```scala
import enumeratum.{PlayJsonEnum, Enum, EnumEntry}

sealed trait MapMode extends EnumEntry
object MapMode extends Enum[MapMode] with PlayJsonEnum[MapMode] {
  val values = findValues
  case object MAP      extends MapMode
  case object LIST     extends MapMode
  case object DOCUMENT extends MapMode
}

  def enumTsType[E <: enumeratum.EnumEntry, AllValues <: enumeratum.Enum[E]: Manifest](enum: AllValues)(
      implicit m: Manifest[AllValues]): TSType[E] =
    TSType.alias[E](
      m.runtimeClass.getSimpleName.stripSuffix("$"),
      TSUnion(enum.values.map(e => TSLiteralString(e.entryName)))
    )
  implicit val myEnumTsType: TSType[MapMode] = enumTsType(MapMode)
```

Directly adding the mapMode enum shown above to the typescriptClassesToGenerateFor SBT key results in a funky code generation of:

```typescript
export interface IDOCUMENT {}
export interface ILIST {}
export interface IMAP {}
export type MapMode = (IDOCUMENT | ILIST | IMAP)
```

but if the enum is simply referenced from a case class that is in the SBT key instead, the code generated is a lot more desirable for me:

```
export type MapMode = ("MAP" | "LIST" | "DOCUMENT")
```

I've created a project based on my use case: https://github.com/tonymichaelson/scala-tsi-issue-74

I'm exporting the enum directly via `typescriptClassesToGenerateFor` and also as a field referenced from a case class. The case class reference produces the desired Typescript with string literals but the direct export does not.

I also noticed if I use `typescriptClassesToGenerateFor` *and also* reference the enum from a case class it will be exported *twice*. I added that to the project as well. I hope it's clear :)

**Comment by dhoepelman** [comment]
Hey @tony. I looked into the problem.

The types you specify in `typescriptClassesToGenerateFor` need to have names, indicated by a `TSNamedType[T]`.
`TSType.alias` and similar return that, but you explicitly annotated the implicits with `TSType[T]` (which doesn't *need* to have a name)

Changing the annotation from `TSType` to `TSNamedType` fixes the problem

```
object TypescriptExports extends DefaultTSTypes {
  def enumTsType[E <: enumeratum.EnumEntry, AllValues <: enumeratum.Enum[E]: Manifest](enum: AllValues)(
    implicit m: Manifest[AllValues]): TSNamedType[E] =
    TSType.alias[E](
      m.runtimeClass.getSimpleName.stripSuffix("$"),
      TSUnion(enum.values.map(e => TSLiteralString(e.entryName)))
    )

  implicit val funkyMapMode: TSNamedType[FunkyMapMode] = enumTsType(FunkyMapMode)
  implicit val mapMode     : TSNamedType[MapMode]      = enumTsType(MapMode)
}
```

"Even" I didn't immediately spot it, so I will look into improving this behavior.

**Comment by tonymichaelson** [comment]
That's great to hear! Thanks for figuring that out. I certainly wouldn't have dug deep enough. I appreciate your effort and I verified it fixed the problem on my end too.

Kind regards,
Tony

**Comment by dhoepelman** [comment]
Fixed in #97 which I'll release with 0.4.0 soon
